### PR TITLE
FLINT 2.6.2

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: F/FLINT/build_tarballs.jl [hunk 1]
```diff
@@ -16,12 +17,8 @@ cd $WORKSPACE/srcdir
 cd flint2/
 if [[ ${target} == *musl* ]]; then
    # because of some ordering issue with pthread.h and sched.h includes
-   export CFLAGS=-D_GNU_SOURCE=1
-   # and properly define _GNU_SOURCE here as well to avoid many warnings
-   sed -i -e 's/#define _GNU_SOURCE$/#define _GNU_SOURCE 1/' thread_pool.h configure
+   #export CFLAGS=-D_GNU_SOURCE
```
Comment (fingolfin) [inline F/FLINT/build_tarballs.jl:20]: I want to see what the exact errors are (resp. if they still are there). If so, we can think about submitting a "proper" fix (or else an issue report) upstream.

**Comment by fingolfin** [comment]
also: ping @thofma @benlorenz so they are aware of this; we need to coordinate this with LoadFlint.jl, Nemo.jl, Polymake.jl, Singular.jl (another reason this PR is marked as "draft" for now)

**Comment by thofma** [comment]
We should not merged this due to https://github.com/wbhart/flint2/issues/803.

File: F/FLINT/build_tarballs.jl [hunk 2]
```diff
@@ -45,4 +41,16 @@ dependencies = [
...
+    #to match the global gmp ones
+    fm = dlsym(libflint_handle, :__flint_set_memory_functions)
+    ccall(fm, Nothing,
+      (Ptr{Nothing},Ptr{Nothing},Ptr{Nothing},Ptr{Nothing}),
+        cglobal(:jl_malloc),
+        cglobal(:jl_calloc),
+        cglobal(:jl_realloc),
+        cglobal(:jl_free))
+  end
+""")
```
Comment (fingolfin) [inline F/FLINT/build_tarballs.jl:56]: With this in, assuming it actually works as intended (I have not yet checked), we could probably get rid of LoadFlint.jl -- at least once we drop support for Julia 1.0-1.2 in our packages (e.g. if/when Julia 1.6 is released as a new LTS).

Comment (thofma) [reply F/FLINT/build_tarballs.jl:56]:
```suggestion
               init_block = """
  if !Sys.iswindows() && !(get(ENV, "NEMO_THREADED", "") == "1")
    #to match the global gmp ones
    fm = dlsym(libflint_handle, :__flint_set_memory_functions)
    ccall(fm, Nothing,
      (Ptr{Nothing},Ptr{Nothing},Ptr{Nothing},Ptr{Nothing}),
        cglobal(:jl_malloc),
        cglobal(:jl_calloc),
        cglobal(:jl_realloc),
        cglobal(:jl_free))
  end
""")
```

**Comment by thofma** [comment]
We should go directly to 2.6.2. If you won't find time, I can push it to the finish line if you want.

**Comment by fingolfin** [comment]
Updated to 2.6.2, should be ready to go now.

**Pull request merged**